### ISSUE TITLE
Move to trash does not work

### ISSUE THREAD

**Issue by Mailaender** [flathub/com.github.git_cola.git-cola#2]
The Python package is probably not using the correct FreeDesktop.org portal that works in a sandbox.

**Comment by Erick555** [comment]
Even [direct trash access](https://github.com/flathub/com.github.git_cola.git-cola/commit/d9aa2ed828321b127fad2825a89011e020b35acc) doesn't help for that?

**Comment by Mailaender** [comment]
No and I don't even know where to start fixing this.

**Comment by bbhtt** [comment]
> The Python package is probably not using the correct FreeDesktop.org portal that works in a sandbox.

It doesn't, it operates [directly on the filesystem](https://github.com/arsenetar/send2trash/blob/e59ddcae988aa4fb1e7f1fefc571831f28627f2c/send2trash/plat_other.py#L212) based on the Freedesktop trash spec. So files go into `~/.var/app/com.github.git_cola.git-cola/data/Trash`. That part seems to working.

I have opened an issue about using the trash portal https://github.com/arsenetar/send2trash/issues/82

**Comment by Erick555** [comment]
> That won't work because it is trying to find the trash can using the $XDG_DATA_HOME variable first which is not set to the "real" trashcan in $HOME inside the sandbox.

with `--filesystem=xdg-data/Trash` the trash in sanbox is at `$XDG_DATA_HOME /Trash` bind mounted to "real" trashcan in $HOME so I don't uderstand why ot doesn't work.

**Comment by bbhtt** [comment]
> --filesystem=xdg-data/Trash

I don't think filesystem does any bind mounts? I can't find in man flatpak run.

**Comment by Erick555** [comment]
Basically all `--filesystem` does is bind mounts host paths to sandbox paths, nothing else. If you do `--filesystem=xdg-download` to binds hosts `~/Downloads` in sandbox etc.

**Comment by bbhtt** [comment]
> > --filesystem=xdg-data/Trash
> 
> I don't think filesystem does any bind mounts? I can't find in man flatpak run.

Ok found it here https://github.com/flatpak/flatpak/blob/7a55b7c49f567bad371ecb18d2ef1833f337065a/common/flatpak-context.c#L2961-L2962, it's not mentioned in the docs.

Not sure then what is happening. Calibri's trash implementation is the exact same as send2trash in that both try to find the trash can first by looking at `$XDG_DATA_HOME`. 

Electron applications will work because they use gio trah, gvfs-trash, kioclient trash and trash-cli. I assume the desktop native ones support the portal backend.

**Comment by Erick555** [comment]
The UI of this app hangs each time I want to open any it repo so I can't test what it does with the trash.

**Comment by bbhtt** [comment]
Hm, looks like it is working as expected? "Delete" deletes it permanently and move to trash moves it to real trash.

Also tried via the python console in the sandbox:

```
>>> from send2trash import send2trash
>>> send2trash('test.txt')
```

moves it to real trash

I didn't realise send2trash wasn't being built and didn't know about the bind mount part. When I made the first comment, I reproduced it with a minimal manifest that didn't have `filesystem=xdg-data/Trash`
